vm/src/builtins/code.rs

```diff
@@ -866,6 +1036,26 @@ impl ToPyObject for bytecode::CodeObject {
     }
 }
 
+/// Validates and parses bytecode bytes into CodeUnit instructions.
+/// Returns None if bytecode is invalid (odd length or contains invalid opcodes).
+fn parse_bytecode(bytecode_bytes: &[u8]) -> Option<Box<[CodeUnit]>> {
```

Comment:
Wouldn't it be more appropriate to return
```rust
Result<Box<[CodeUnit]>, MarshalError>
```

Reply:
Maybe marshal is not the appropriate error, I initially thought of marshal because it has `MarshalError::InvalidBytecode`